modules/objdetect/include/opencv2/objdetect.hpp

```diff
@@ -763,29 +763,68 @@ class CV_EXPORTS_W QRCodeEncoder {
-     of the scheme 1:1:3:1:1 according to QR code standard.
+    /** @brief Methods for detecting QR codes: CONTOURS_BASED, ARUCO_BASED */
+    enum Method {
+        CONTOURS_BASED, ///< QR detect method with detect finder pattern as contours
+        ARUCO_BASED ///< QR detect method, with detect finder pattern as ArUco marker
+        // TODO: add WECHAT method
+    };
+
+    QRDetectorParameters();
+
+    CV_PROP_RW Method qrDetectMethod;
+
+    /** @brief The minimum allowed pixel size of a QR module in the smallest image in the image pyramid, default 4.f
+     *
+     * Only ARUCO_BASED parameter.
     */
-    CV_WRAP void setEpsX(double epsX);
-    /** @brief sets the epsilon used during the vertical scan of QR code stop marker detection.
-     @param epsY Epsilon neighborhood, which allows you to determine the vertical pattern
-     of the scheme 1:1:3:1:1 according to QR code standard.
-     */
-    CV_WRAP void setEpsY(double epsY);
+    CV_PROP_RW float minModuleSizeInPyramid;
 
-    /** @brief use markers to improve the position of the corners of the QR code
+    /** @brief The maximum allowed relative rotation for finder patterns in the same QR code, default pi/12
      *
-     * alignmentMarkers using by default
-     */
-    CV_WRAP void setUseAlignmentMarkers(bool useAlignmentMarkers);
+     * Only ARUCO_BASED parameter
+    */
+    CV_PROP_RW float maxRotation;
+
+    /** @brief The maximum allowed relative mismatch in module sizes for finder patterns in the same QR code, default 1.75f
+     *
+     * Only ARUCO_BASED parameter
+    */
+    CV_PROP_RW float maxModuleSizeMismatch;
 
+    /** @brief The maximum allowed module relative mismatch for timing pattern module, default 2.f
+     *
+     * Only ARUCO_BASED parameter
+     * If relative mismatch of timing pattern module more this value, penalty points will be added.
+     * If a lot of penalty points are added, QR code will be rejected. */
+    CV_PROP_RW float maxTimingPatternMismatch;
+
+    /** @brief The maximum allowed percentage of penalty points out of total pins in timing pattern, default 0.4f
+     *
+     * Only ARUCO_BASED parameter
+    */
+    CV_PROP_RW float maxPenalties;
+
+    /** @brief The maximum allowed relative color mismatch in the timing pattern, default 0.2f
+     *
+     * Only ARUCO_BASED parameter
+    */
+    CV_PROP_RW float maxColorsMismatch;
+
+    /** @brief The algorithm find QR codes with almost minimum timing pattern score and minimum size, default 0.9f
+     *
+     * Only ARUCO_BASED parameter
+     *
+     * The QR code with the minimum "timing pattern score" and minimum "size" is selected as the best QR code.
+     * If for the current QR code "timing pattern score" * scaleTimingPatternScore < "previous timing pattern score" and "size" < "previous size", then
+     * current QR code set as the best QR code. */
+    CV_PROP_RW float scaleTimingPatternScore;
+};
+
+class CV_EXPORTS_W CommonQRCodeDetector {
```

Comment:
> CommonQRCodeDetector

`QRCodeDetectorBase`

It is a mess to type "Common" prefix for Intellisense.

Reply:
fixed